app/src/main/java/com/vultisig/wallet/ui/models/VaultAccountsViewModel.kt

```diff
@@ -40,64 +45,134 @@ internal data class AccountUiModel(
     val assetsSize: Int = 0,
 )
 
+@Parcelize /*draggable list requires parcelable data*/
+internal data class ItemAccountUiModel(
+    val addressId: String,
+    val chainName: String,
+    @DrawableRes val logo: Int,
+    val address: String,
+    val nativeTokenAmount: String?,
+    val fiatAmount: String?,
+    val assetsSize: Int = 0,
+) : Parcelable
+
 @HiltViewModel
 internal class VaultAccountsViewModel @Inject constructor(
+    private val navigator: Navigator<Destination>,
+
     private val addressToUiModelMapper: AddressToUiModelMapper,
     private val fiatValueToStringMapper: FiatValueToStringMapper,
 
     private val vaultRepository: VaultRepository,
     private val accountsRepository: AccountsRepository,
+    private val chainsOrderRepository: ChainsOrderRepository,
 ) : ViewModel() {
-    var vault: Vault? by mutableStateOf(null)
+    private var vaultId: String? = null
 
     val uiState = MutableStateFlow(VaultAccountsUiModel())
 
     private var loadVaultNameJob: Job? = null
     private var loadAccountsJob: Job? = null
-
+    private var reIndexJob: Job? = null
     fun loadData(vaultId: String) {
+        this.vaultId = vaultId
         loadVaultName(vaultId)
-        loadAccounts(vaultId)
+        loadAccounts(
+            vaultId = vaultId,
+            showRefreshing = false,
+        )
     }
 
     fun refreshData() {
+        loadAccounts(
+            vaultId = requireNotNull(vaultId),
+            showRefreshing = true,
+        )
+    }
+
+    fun openAccount(account: ItemAccountUiModel) {
+        val vaultId = vaultId ?: return
+        val chainId = account.addressId
+
         viewModelScope.launch {
-            uiState.update { it.copy(isRefreshing = true) }
-            // TODO: add refresh logic here
-            delay(3000)
-            uiState.update { it.copy(isRefreshing = false) }
+            navigator.navigate(
+                Destination.ChainTokens(
+                    vaultId = vaultId,
+                    chainId = chainId,
+                )
+            )
         }
     }
 
     private fun loadVaultName(vaultId: String) {
         loadVaultNameJob?.cancel()
         loadVaultNameJob = viewModelScope.launch {
             val vault = requireNotNull(vaultRepository.get(vaultId))
-            this@VaultAccountsViewModel.vault = vault
             uiState.update { it.copy(vaultName = vault.name) }
         }
     }
 
-    private fun loadAccounts(vaultId: String) {
+    private fun loadAccounts(
+        vaultId: String,
+        showRefreshing: Boolean,
+    ) {
```

Comment:
Again, these changes are from some of my PR's, please do something with git history